app/assets/javascripts/components/sidebar/posts_main_component.js.coffee

```diff
@@ -0,0 +1,44 @@
+# @state.posts is an array of json objects describing posts
+window.PostsMainComponent = React.createClass
+	getInitialState: ->
+		{posts_json: []}
+
+	componentWillMount: ->
+		@loadPostsFromServer()
+	
+	componentDidMount:->
+		@pollServer()	
+
+	loadPostsFromServer: ->
+		console.log "loading posts from server"
+		url = 'posts' + (if @props.postsAreRelevant then '/relevant' else '')
+		return $.ajax
+			url: url+ '.json'
+			dataType: 'json'
+			cache: false
+			success: (posts) =>
+				@setState {posts_json: posts}
```

Comment:
just `posts` not `posts_json`...you're working in javascript so it's obvious that a `post` should be a javascript object.